fwd.go

```diff
@@ -136,26 +136,30 @@ func LocalForward(state *State, s *stack.Stack, conn KaConn, targetAddr net.Addr
 			srcIP    net.Addr
 			ppPrefix = ""
 		)
+		// When doing local forward, if the source IP of host
+		// connection had routable IP (unlike
+		// 127.0.0.1)... well... spoof it! The client might find it
+		// useful who launched the connection in the first place.
 		if proxyProtocol == false {
-			// When doing local forward, if the source IP of host
-			// connection had routable IP (unlike
-			// 127.0.0.1)... well... spoof it! The client might find it
-			// useful who launched the connection in the first place.
-			raddr := conn.RemoteAddr()
-			if IPNetContains(state.StaticRoutingDeny, netAddrIP(raddr)) == false {
-				srcIP = raddr
-			}
+			srcIP = conn.RemoteAddr()
```

Comment:
@joel0 I don't get this. The intention here is to send traffic into guest, but make sure the guest can respond. If we will send traffic sourced from 127.0.0.1 the guest will not be able to respond - since it will tx the responses into 127.0.0.1. The same for 10.0.2.X IP's, the response will never get back to us. Why did you tweak this?

Reply:
The source IP is spoofed whenever possible, but when IPv4->IPv6 or IPv6->IPv4 the source IP can't be preserved (see in-source comments for why).  IPv4<->IPv6 currently only works when srcIP is in StaticRoutingDeny.

So, what should be done with those forwards?
* Drop them? (with a better error message than "connect udp 10.0.2.100:10010: network is unreachable"?)
* Forward them?
  * With no source IP?
  * Ignore targetAddr and replace it with 10.0.2.100/2001:2::100 depending on srcIP's IP version?

Reply:
Right, so keep the current code and add another thing to if. if protocols don't match (v4 to v6), then don't spoof. Otherwise keep the logic as it is

Reply:
I don't exactly understand what you want.  Unless I'm missing something, I don't believe any logic was changed outside of v4<->v6, other than simplifying the code.

Are you suggesting that lines 149-162 (new line numbers) should be moved to 145 and copied to 150 (old line numbers)?  I don't mind making a commit to do that, but I want to understand what you want before I make a guess at changing the code.

Reply:
Follow up from our call: see line 153 which checks StaticRoutingDeny and then sets srcIP = nil.  This should result in the same thing as what I removed.